src/components/data/row/CurrencyRow.tsx

```diff
@@ -22,21 +34,48 @@ interface Props {
 // A view representing the data from a wallet, used for rows, cards, etc.
 // -----------------------------------------------------------------------------
 const CurrencyRowComponent = (props: Props) => {
-  const { marginRem, showRate = false, token, tokenId, wallet } = props
+  const {
+    //
+    customAsset,
+    marginRem,
+    showRate = false,
+    token,
+    tokenId
+  } = props
+  let { wallet } = props
+  const {
+    displayCurrencyCode,
+    displayName,
+    nativeAmount: customAssetBalance,
+    referenceTokenId,
+    wallet: customAssetWallet
+  } = customAsset ?? {
+    displayCurrencyCode: undefined,
+    displayName: undefined,
+    nativeAmount: undefined,
+    referenceTokenId: undefined,
+    wallet: undefined
+  }
+  wallet = customAssetWallet ?? wallet
 
   // Currency code and wallet name for display:
-  const allTokens = wallet.currencyConfig.allTokens
-  const tokenFromId = token != null ? token : tokenId == null ? null : allTokens[tokenId]
-  const { currencyCode } = tokenFromId == null ? wallet.currencyInfo : tokenFromId
-  const name = useWalletName(wallet)
+  const allTokens = wallet.currencyConfig != null ? wallet.currencyConfig.allTokens : null
+  const tokenFromId = customAsset != null || allTokens == null ? null : token != null ? token : tokenId == null ? null : allTokens[tokenId]
+  const { currencyCode } = displayCurrencyCode != null ? { currencyCode: displayCurrencyCode } : tokenFromId == null ? wallet.currencyInfo : tokenFromId
+  const walletName = useWalletName(wallet)
+  const name = displayName ?? walletName
```

Comment:
I gave a stab at cleaning all this up in a fixup commit (I also fixed the snapshot test failure).